### PR TITLE
Add radio/checkbox serialization to forms

### PR DESCRIPTION
**Pull request by PartMan7**
These changes allow for submitting data from radio/checkbox elements in form's data-submitsend.

### CODE REVIEW

**Comment by Zarel** [comment]
I don't understand what the `checkboxes` table is for. It seems completely unnecessary. The `radios` table also seems overcomplicated.

File: js/client.js [hunk 1]
```diff
@@ -835,12 +835,25 @@ function toId() {
 			// querySelector dates back to IE8 so we can use it
 			// fortunate, because form serialization is a HUGE MESS in older browsers
 			var elements = form.querySelectorAll('input[name], select[name], textarea[name], keygen[name]');
-			var out = [];
+			var out = [], checkboxes = {}, radios = {};
 			for (var i = 0; i < elements.length; i++) {
 				var element = elements[i];
-				// TODO: values are a mess in the DOM; checkboxes/select probably need special handling
```
Comment (Zarel) [inline js/client.js:841]: You deleted this comment but you don't seem to have handled the `select` element.

File: js/client.js [hunk 2]
```diff
@@ -835,12 +835,25 @@ function toId() {
...
-				out.push([element.name, element.value]);
-			}
+				if (element.type === 'radio') {
+					var name = ' ' + element.name;
+					if (element.checked) radios[name] = element.value;
+					else if (!radios[name]) radios[name] = false;
+				} else if (element.type === 'checkbox') {
+					if (!checkboxes[' ' + element.name]) checkboxes[' ' + element.name] = [];
+					if (element.checked) checkboxes[' ' + element.name].push(element.value);
```
Comment (Zarel) [inline js/client.js:847]: Oh, I see, you're trying to handle the case where multiple checkboxes have the same name.

Is that actually how the spec handles it?

Comment (PartMan7) [reply js/client.js:847]: Okay I have no idea what I was thinking when I wrote the checkboxes, sorry
Lemme rewrite this, gimme a minute

**Comment by PartMan7** [comment]
Wait, doesn't select already return the selected value as element.value?

**Comment by Zarel** [comment]
I didn't know. Anyway, while I'm sure the stuff you wrote is _useful_, none of it belongs in this file.

File: js/client.js [hunk 3]
```diff
@@ -835,12 +835,21 @@ function toId() {
 			// querySelector dates back to IE8 so we can use it
 			// fortunate, because form serialization is a HUGE MESS in older browsers
 			var elements = form.querySelectorAll('input[name], select[name], textarea[name], keygen[name]');
-			var out = [];
+			var out = [], radios = {};
```
Comment (Zarel) [inline js/client.js:838]:
```suggestion
			var out = [];
```

**Comment by Zarel** [comment]
No hard feelings, but 6383b4adc4bb4fabd is just a much simpler and more spec-compliant implementation.

I do recognize what you were trying to do with checkboxes now: support having multiple with the same name. But that's not how forms are usually serialized, so you should be doing that inside `submitSend` rather than inside `serializeForm`.

(Although I don't think it's really necessary in the first place. It's easy enough to write your commands not to need such a niche feature.)

**Pull request closed without merging**